core/arch/arm/plat-rockchip/conf.mk

```diff
@@ -1,18 +1,52 @@
 PLATFORM_FLAVOR ?= rk322x
 
-ifeq ($(PLATFORM_FLAVOR),rk322x)
-include ./core/arch/arm/cpu/cortex-a7.mk
-$(call force,CFG_TEE_CORE_NB_CORE,4)
-endif
-
 $(call force,CFG_GENERIC_BOOT,y)
 $(call force,CFG_GIC,y)
-$(call force,CFG_HWSUPP_MEM_PERM_PXN,y)
 $(call force,CFG_PM_STUBS,y)
 $(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
-$(call force,CFG_PSCI_ARM32,y)
-$(call force,CFG_BOOT_SECONDARY_REQUEST,y)
 $(call force,CFG_8250_UART,y)
 
+CFG_DT ?= y
 CFG_WITH_STACK_CANARIES ?= y
 CFG_WITH_STATS ?= y
+CFG_NUM_THREADS ?= 4
+
+ifeq ($(PLATFORM_FLAVOR),rk322x)
+include ./core/arch/arm/cpu/cortex-a7.mk
+$(call force,CFG_TEE_CORE_NB_CORE,4)
+$(call force,CFG_PSCI_ARM32,y)
+$(call force,CFG_BOOT_SECONDARY_REQUEST,y)
+
+CFG_TZDRAM_START ?= 0x68400000
+CFG_TZDRAM_SIZE ?= 0x00200000
+CFG_SHMEM_START ?= 0x68600000
+CFG_SHMEM_SIZE ?= 0x00100000
+endif
+
+ifeq ($(PLATFORM_FLAVOR),rk3399)
+include core/arch/arm/cpu/cortex-armv8-0.mk
+$(call force,CFG_TEE_CORE_NB_CORE,6)
+$(call force,CFG_ARM_GICV3,y)
+
+CFG_TZDRAM_START ?= 0x30000000
+CFG_TZDRAM_SIZE  ?= 0x02000000
+CFG_SHMEM_START  ?= 0x32000000
+CFG_SHMEM_SIZE   ?= 0x00400000
+endif
+
+ifeq ($(PLATFORM_FLAVOR),px30)
+include core/arch/arm/cpu/cortex-armv8-0.mk
+$(call force,CFG_TEE_CORE_NB_CORE,4)
+
+CFG_TZDRAM_START ?= 0x30000000
+CFG_TZDRAM_SIZE  ?= 0x02000000
+CFG_SHMEM_START  ?= 0x32000000
+CFG_SHMEM_SIZE   ?= 0x00400000
+endif
+
+ifeq ($(platform-flavor-armv8),1)
+$(call force,CFG_ARM64_core,y)
+$(call force,CFG_WITH_ARM_TRUSTED_FW,y)
+$(call force,CFG_WITH_LPAE,y)
```

Comment:
minor: this is not needed. 64bit implies LPAE.

Reply:
Hmm, it doesn't seem to be that way. Removing that option results in a compile error because optee then seems to want to compile core_mmu_v7.c if CFG_WITH_LPAE is not set.

Similarly, the only option forcing CFG_WITH_LPAE would be CFG_CORE_LARGE_PHYS_ADDR (see core/arch/arm/arm.mk), but even this is not always defined for armv8 it seems?

So in short it is not implied, it seems.

Reply:
Sorry, yes, you're right.
Better keep it.